docker/snippets/rosdep.dockerfile

```diff
@@ -12,16 +12,29 @@ RUN apt-get update && \
 ARG ROS_DISTRO
 ARG ROS_WORKSPACE="./workspace"
 ARG ROSDEP_METAPACKAGE=""
+ARG ROSDEP_OS_NAME="ubuntu"
+ARG ROSDEP_OS_VERSION="jammy"
+ARG ROSDEP_SKIP_KEYS=""
 ARG ROS_INSTALL_PREFIX="/opt/ros/${ROS_DISTRO}/"
 COPY ${ROS_WORKSPACE}/src /tmp/workspace/src
 RUN . ${ROS_INSTALL_PREFIX}/setup.sh && \
     cd /tmp/workspace && \
     apt-get update && \
-    if [ -z "$ROSDEP_METAPACKAGE" ]; then \
+    if [[ -z "${ROSDEP_METAPACKAGE}" ]]; then \
         ROSDEP_FROM_PATHS="src"; \
     else \
-        ROSDEP_FROM_PATHS=$(colcon list --packages-up-to $ROSDEP_METAPACKAGE | awk '{print $2}' | tr '\n' ' '); \
+        ROSDEP_FROM_PATHS=$(colcon list --packages-up-to ${ROSDEP_METAPACKAGE} | awk '{print $2}' | tr '\n' ' '); \
     fi && \
-    rosdep install --from-paths $ROSDEP_FROM_PATHS --ignore-src -r -y && \
+    rosdep install --from-paths ${ROSDEP_FROM_PATHS} --ignore-src -r -y --skip-keys="${ROSDEP_SKIP_KEYS}" --os=${ROSDEP_OS_NAME}:${ROSDEP_OS_VERSION} --rosdistro=${ROS_DISTRO} && \
```

Comment:
Explanation: added the skip keys that i explained above, explicitly set the os version to look at for rosdep (would be focal, which isn't what we want, if not specified) and explicitly set the rosdistro, which I don't think is actually needed but whatever.